packages/webview_flutter/tizen/src/webview.cc

```diff
@@ -518,128 +518,112 @@ static LWE::KeyValue KeyNameToKeyValue(const std::string& ecore_key,
-  } else if (ecore_key == "XF86Sleep") {
+  } else if (key == "XF86Sleep") {
     return LWE::KeyValue::TVSleep;
-  } else if (ecore_key == "XF86Caption") {
+  } else if (key == "XF86Caption") {
     return LWE::KeyValue::TVCaption;
-  } else if (ecore_key == "XF86More") {
+  } else if (key == "XF86More") {
     return LWE::KeyValue::TVMore;
-  } else if (ecore_key == "XF86BTVoice") {
+  } else if (key == "XF86BTVoice") {
     return LWE::KeyValue::TVBTVoice;
-  } else if (ecore_key == "XF86Color") {
+  } else if (key == "XF86Color") {
     return LWE::KeyValue::TVColor;
-  } else if (ecore_key == "XF86PlayBack") {
+  } else if (key == "XF86PlayBack") {
     return LWE::KeyValue::TVPlayBack;
   }
-  LOG_WARN("Unknown key name: %s", ecore_key.c_str());
+  LOG_WARN("Unknown key name: %s", key.c_str());
   return LWE::KeyValue::UnidentifiedKey;
 }
 
-void WebView::DispatchKeyDownEvent(Ecore_Event_Key* key_event) {
+bool WebView::SendKey(const char* key, const char* string, const char* compose,
+                      uint32_t modifiers, uint32_t scan_code, bool is_down) {
   if (!IsFocused()) {
-    return;
+    return false;
   }
 
-  std::string key_name = key_event->keyname;
-  bool is_shift_pressed = key_event->modifiers & 1;
+  bool is_shift_pressed = modifiers & 1;
 
   struct Param {
     LWE::WebContainer* webview_instance;
     LWE::KeyValue key_value;
+    bool is_down;
   };
-  Param* param = new Param();
-  param->webview_instance = webview_instance_;
-  param->key_value = KeyNameToKeyValue(key_name, is_shift_pressed);
-
-  webview_instance_->AddIdleCallback(
-      [](void* data) {
-        Param* param = reinterpret_cast<Param*>(data);
-        param->webview_instance->DispatchKeyDownEvent(param->key_value);
-        param->webview_instance->DispatchKeyPressEvent(param->key_value);
-        delete param;
-      },
-      param);
-}
 
-void WebView::DispatchKeyUpEvent(Ecore_Event_Key* key_event) {
-  if (!IsFocused()) {
-    return;
-  }
-
-  std::string key_name = key_event->keyname;
-  bool is_shift_pressed = key_event->modifiers & 1;
-
-  struct Param {
-    LWE::WebContainer* webview_instance;
-    LWE::KeyValue key_value;
-  };
   Param* param = new Param();
   param->webview_instance = webview_instance_;
-  param->key_value = KeyNameToKeyValue(key_name, is_shift_pressed);
+  param->key_value = KeyToKeyValue(key, is_shift_pressed);
+  param->is_down = is_down;
 
   webview_instance_->AddIdleCallback(
       [](void* data) {
         Param* param = reinterpret_cast<Param*>(data);
-        param->webview_instance->DispatchKeyUpEvent(param->key_value);
+        if (param->is_down) {
+          param->webview_instance->DispatchKeyDownEvent(param->key_value);
+          param->webview_instance->DispatchKeyPressEvent(param->key_value);
+        } else {
+          param->webview_instance->DispatchKeyUpEvent(param->key_value);
+        }
         delete param;
       },
       param);
+
+  return false;
```

Comment:
I think we've discussed about adding a `CanGoForward` check on a back key press and invoking `GoBack` if the result is true. Are you going to implement that later?

Reply:
Thanks!